forc/src/ops/forc_new.rs

```diff
@@ -0,0 +1,190 @@
+use crate::cli::NewCommand;
+use crate::utils::{
+    defaults,
+    program_type::{ProgramType, ProgramType::*},
+};
+use anyhow::Result;
+use forc_util::{print_light_blue, print_light_green, validate_name};
+use serde::Deserialize;
+use std::fs;
+use std::path::Path;
+use sway_utils::constants;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum FileType {
+    File,
+    Dir,
+}
+
+// Dead fields required for deserialization.
+#[allow(dead_code)]
+#[derive(serde::Deserialize, Debug)]
+struct Links {
+    git: String,
+    html: String,
+    #[serde(rename = "self")]
+    cur: String,
+}
+
+// Dead fields required for deserialization.
+#[allow(dead_code)]
+#[derive(serde::Deserialize, Debug)]
+struct ContentResponse {
+    #[serde(rename = "_links")]
+    links: Links,
+    download_url: Option<String>,
+    git_url: String,
+    html_url: String,
+    name: String,
+    path: String,
+    sha: String,
+    size: u64,
+    #[serde(rename = "type")]
+    file_type: FileType,
+    url: String,
+}
+
+#[allow(dead_code)]
+#[derive(serde::Deserialize)]
+struct GithubRepoResponse {
+    sha: String,
+    url: String,
+    // We only care about the tree here
+    tree: Vec<GithubTree>,
+    truncated: bool,
+}
+
+#[allow(dead_code)]
+#[derive(serde::Deserialize)]
+struct GithubTree {
+    mode: String,
+    // We only care about the "path" which are files / directory names
+    path: String,
+    sha: String,
+    size: Option<usize>,
+    #[serde(rename = "type")]
+    data_type: String,
+    url: String,
+}
```

Comment:
We should be able to remove all types and any logic related to GitHub requests/responses in this module after removing the unnecessary `--template` option. The need to add `#[allow(dead_code)]` here is a good indicator these are no longer needed :)